hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/BoundedWriteBehindQueue.java

```diff
@@ -149,26 +154,9 @@ public void clear() {
         return queue.asList();
     }
 
-    /**
-     * Adds all elements to the supplied collection which are smaller than or equal to the given time.
-     *
-     * @param time       given time.
-     * @param collection to add filtered elements.
-     */
-    @Override
-    public void getFrontByTime(long time, Collection<E> collection) {
-        queue.getFrontByTime(time, collection);
-    }
-
-    /**
-     * Adds some number of elements to the supplied collection starting from the head of this queue.
-     *
-     * @param numberOfElements number of elements to add.
-     * @param collection       to add filtered elements.
-     */
     @Override
-    public void getFrontByNumber(int numberOfElements, Collection<E> collection) {
-        queue.getFrontByNumber(numberOfElements, collection);
+    public void filter(IPredicate<E> predicate, Collection<E> collection) {
```

Comment:
It is already there, on the interface, please see https://github.com/hazelcast/hazelcast/pull/7428/files#diff-79bbf4022aca33f09a86c90f0160993aR98